core/pta/attestation.c

```diff
@@ -55,9 +55,11 @@ static TEE_Result generate_key(void)
 
 	res = allocate_key();
 	if (res)
-		return res;
+		goto err;
```

Comment:
Not needed, if `allocate_key()` failed there should be nothing to free.

Reply:
In `allocate_key()`, `crypto_acipher_alloc_rsa_keypair()` could fail but `key` still needs to be freed no?

Reply:
You're right. The `goto err;` is correct here.